decode/src/main.rs

```diff
@@ -144,12 +160,83 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
             println!("{:#?}", ProteusEnvelope::from(message));
             Ok(())
         }
-        Command::MlsMessage { message } => {
+        Command::MlsMessage {
+            message,
+            raw_message,
+            members,
+            identities,
+        } => {
             let input: String = message.contents()?;
             let bytes = base64::prelude::BASE64_STANDARD.decode(input)?;
-            let message = MlsMessageIn::tls_deserialize(&mut bytes.as_slice())?;
-            println!("{message:#?}");
+
+            if let Ok(message) = MlsMessage::mls_decode(&mut bytes.as_slice()) {
+                if raw_message {
+                    println!("{message:#?}");
+                }
+
+                if let Some(group_info) = message.into_group_info() {
+                    if members {
+                        group_info.print_members();
+                    }
+
+                    if identities {
+                        group_info.print_basic_identities();
+                    }
+                }
+            }
+
+            if let Ok(group_info) = GroupInfo::mls_decode(&mut bytes.as_slice()) {
+                if raw_message {
+                    println!("{group_info:#?}");
+                }
+
+                if members {
+                    group_info.print_members();
+                }
+
+                if identities {
+                    group_info.print_basic_identities();
+                }
+            }
+
             Ok(())
         }
     }
 }
+
+trait GroupInfoExt {
+    fn print_members(&self);
+    fn print_basic_identities(&self);
+}
+
+impl GroupInfoExt for GroupInfo {
+    fn print_members(&self) {
```

Comment:
I'm not sure why we need a trait here, why not just have regular functions that take `&GroupInfo`, e.g.
```rust
fn print_members(group_info: &GroupInfo) { .. }
```
?